src/main/java/it/reply/orchestrator/service/VaultServiceImpl.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright © 2019 I.N.F.N.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package it.reply.orchestrator.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import it.reply.orchestrator.config.properties.VaultProperties;
+import it.reply.orchestrator.exception.VaultTokenExpiredException;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.stereotype.Service;
+import org.springframework.vault.VaultException;
+import org.springframework.vault.authentication.TokenAuthentication;
+import org.springframework.vault.client.VaultEndpoint;
+import org.springframework.vault.core.VaultTemplate;
+import org.springframework.vault.support.VaultResponse;
+
+@Service
+@EnableConfigurationProperties(VaultProperties.class)
+public class VaultServiceImpl implements VaultService {
+
+  @Autowired
+  private VaultProperties vaultProperties;
+
+  private String token;
```

Comment:
Token should not be a field. `VaultServiceImpl` is a Spring bean with default scope; therefore it is a singleton and it's shared by all request coming from different users